### ISSUE TITLE
As a user I want to send stickers and set up my own integrations

### ISSUE THREAD

**Issue by lampholder** [vector-im/user-stories#6]
- I want it to be easy to set up integrations and send stickers, so I want this setting to be 'on' by default
- The first time I send a sticker or set up my own bot, bridge or widget I use an integration manager I will be prompted to read and accept the terms of service
    - If I refuse to accept the terms, I should understand that I won't be able to send stickers or set up my own integrations
    - If the integration manager materially changes its terms of service I will be prompted to review and accept these terms again
- If I have elected to use an integration manager to set up integrations and send stickers, clicking on the integration manager launcher or sticker picker will:
    - prompt me to accept the integration manager's terms if it has terms that I have not accepted
    - show me the integration manager/sticker picker if there are no terms need accepting
- If I have elected **not** to use an integration manager to set up integrations and send stickers, clicking on the integration manager launcher or sticker picker will:
    - advise me that I have disabled this feature and can reenable it in user settings
- My choice of whether or not to use an integration manager to set up integrations and send stickers is persisted and honoured across all Riot clients/sessions
    - If I change my mind, I can just toggle this setting in User Settings

### Edge cases:

- My making this choice indicates that I wish to use an integration manager to set up integrations and send stickers - it is independent of choosing _which_ integration manager I am using
    - If I am using whichever integration manager is suggested by my service provider and my service provider changes the suggested integration manager (perhaps by changing the value in .well-known), if I have chosen to use an integration manager then I will start using the new integration manager as soon as my client has picked up those changes (modulo signing any new terms of service with the new integration manager)
- If I (somehow) have no integration manager configured (there's nothing in .well-known, config.json or account data), the integration manager button and sticker picker should still be clickable but pop up a prompt telling you that you do not have an integration manager configured
    - *it is known that in this case I will not be able to use scalar widgets* (they will render as `FORBIDDEN`)

**Comment by lampholder** [comment]
N.B. The is different from https://github.com/matrix-org/matrix-doc/pull/2315 which allows users to set 'none' as their integration manager. The intention here is to give users clear control over their interaction with integration managers by separating in users' minds two distinct concepts:
1. using an integration manager to set up integrations and send stickers
1. using a widget that happens to have been provisioned using an integration manager

It is an important distinction - the UX should not present a choice for you to 'disconnect from the integration manager' as it does for the identity server.

If a user does not want to share their data with an integration manger, they should:
- **elect not to use an integration manager for setting up integrations and sending stickers**
- **choose not to give any widgets managed by that integration manger permission to load**

If anybody is reading this wondering why we didn't just treat these two as one, it's because widgets can come from a range of sources (different integration mangers, or no integration manager at all) and toggling off all widgets along with integration-managers-for-setup-and-stickers felt like an overly blunt instrument.

This approach lets users choose not use to an integration manager for sending stickers or setting up their own integrations, but lets provision widgets _without_ an integration manager (using the `/addwidget` command, and still use widgets provisioned by other people (with or without an integration manager).

**Comment by turt2live** [comment]
> My choice of whether or not to use an integration manager to set up integrations and send stickers is persisted and honoured across all Riot clients/sessions

This is https://github.com/matrix-org/matrix-doc/pull/2315 (the spec should be in support of the UX described here - if it's not, please leave a comment on the MSC)

**Comment by lampholder** [comment]
N.B. The is different from https://github.com/matrix-org/matrix-doc/pull/2315 which allows users to set 'none' as their integration manager.

_This_ option lets users say they don't want to use an integration manager for sending stickers or setting up their own integrations, but lets them still use widgets provisioned by other people using an integration manager.

Due to scalar's reliance upon scalar tokens being passed to it to render any widget, riots with an integration manager set to 'none' cannot render scalar widgets provisioned by other users.

In order to completely avoid sharing personal data with a given integration manager, the user should:
- either:
    - configure riot not to use an integration manager for setting up integrations or sending stickers, or
    - just not use it to do those things
- and:
    - not load widgets which are hosted by that integration manager

**Comment by lampholder** [comment]
Implementation is blocked on approval of https://github.com/vector-im/riot-meta/pull/310

**Comment by turt2live** [comment]
The MSC mentioned above should be ignored - we don't need it.

**Comment by lampholder** [comment]
Potential Mobile UX:

### Integrations

**I want to send stickers and set up integrations.** &nbsp;&nbsp;&nbsp;     ⊂⦿
This requires the use of an integration manager. Setting up integrations will share configuration data with your client's integration manager: https://scalar.vector.im.

**Comment by lampholder** [comment]
Story modified to reflect Nad's statement that we **should never hide UI**.

**Comment by lampholder** [comment]
@nadonomy - this is the issue with outstanding wordsmithing

**Comment by BillCarsonFr** [comment]
> If I (somehow) have no integration manager configured (there's nothing in .well-known, config.json or account data), the integration manager button and sticker picker should be hidden exaxctly as if I had chosen not to use an integration manager to set up integrations or send stickers 

@lampholder Didn't we say that we should let the button visible for discovery even if they do not work?

**Comment by lampholder** [comment]
> @lampholder Didn't we say that we should let the button visible for discovery even if they do not work?

We did - I updated the top of the issue but forgot there was another reference here. @nadonomy we do need a bit more copy for the 'no integration manager configured' case. Probably something like:

"Your client has not been set up to use an integration manager - you need an integration manager to set up and configure integrations. Please talk to your service provider for more information."

**Comment by turt2live** [comment]
@lampholder to clarify, this affects the integration manager button *and* the sticker picker button? Did that copy get approved?

**Comment by nadonomy** [comment]
When clicking on the 'Manage Integrations' or sticker picker buttons in Riot, we should use the following copy in modals:

If Riot can't use an Integration Manager (due to synapse or riot config):

> "Your Riot doesn't allow you to use an Integration Manager to do this. Please contact an admin."
> **[OK]**

If the user has toggled off 'Manage integrations' in Settings

> "Enable 'Manage Integrations' in Settings to do this."
> [OK] **[Settings]**

**Comment by lampholder** [comment]
I don't believe this has any outstanding design issues, so I'm removing `needs-ux-design`
